source/scpd/Cpp.d

```diff
@@ -348,6 +352,28 @@ extern(C++, (StdNamespace)) extern(C++, class) struct vector (T, Alloc = allocat
                 vec.push_back(item);
             return vec;
         }
+
+        public void serialize (scope SerializeDg dg) const @safe
+        {
+            serializePart(this.length, dg);
+            foreach (ref entry; this.constIterator())
+                serializePart(entry, dg);
+        }
+
+        static QT fromBinary (QT) (scope DeserializeDg data,
+            scope const ref DeserializerOptions opts) @safe
+        {
+            import std.traits;
+            import scpd.types.Utils;
+
+            auto arr = deserializeFull!(QT.ElementType[])(data, opts);
```

Comment:
Hmm.. Why is it allocating an array here? Instead of manually walking over the serialized data?

Reply:
Right, could do that. I think the original concern was to make sure the data would stay alive after this returns. I'll see if I can improve this.

Honestly this is the worst part of the code, because dealing with vectors is terrible. We really need a way to build a `const` vector from an array.

Reply:
Ah right because the vector only stores the pointers. Then if there's no better way, I'll just approve this.